### ISSUE TITLE
Compilation error following the steps

### ISSUE THREAD

**Issue by fernandorrmartins** [otland/forgottenserver#4420]
Trying to compile following the steps in this repository, i got an error.
SS of followed steps:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/32397071/233841080-56a8c744-22f5-40b3-9f2b-72dab802e715.png">

The issue:
`root@ubuntu20:/opt/forgottenserver# mkdir build && cd build
root@ubuntu20:/opt/forgottenserver/build# cmake ..
-- The CXX compiler identification is GNU 9.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find cryptopp (missing: cryptopp_DIR)
-- Found Crypto++: /usr/include
-- Could NOT find unofficial-libmariadb (missing: unofficial-libmariadb_DIR)
-- MySQL Include dir: /usr/include/mysql  library dir: /usr/lib/x86_64-linux-gnu
-- MySQL client libraries: /usr/lib/x86_64-linux-gnu/libmysqlclient.so;-ldl
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found LuaJIT: /usr/lib/x86_64-linux-gnu/libluajit-5.1.so;/usr/lib/x86_64-linux-gnu/libm.so (found version "2.1.0-beta3")
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake (found suitable version "1.71.0", minimum required is "1.66.0") found components: system iostreams
-- IPO / LTO enabled
-- Configuring done
CMake Error at CMakeLists.txt:76 (add_executable):
  Target "tfs" links to target "pugixml::pugixml" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?

CMake Error at src/CMakeLists.txt:172 (add_library):
  Target "tfslib" links to target "pugixml::pugixml" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
`

Thie problem happened in Ubuntu 20.

**Comment by jonaspcarvalho** [comment]
Hello everyone,

I also had troubles compiling the project in two different Ubuntu releases.

First on **Ubuntu 18.04**:

```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.6 LTS
Release:	18.04
Codename:	bionic
```

```
../forgottenserver/build# cmake ..
-- Could NOT find cryptopp (missing: cryptopp_DIR)
-- Could NOT find unofficial-libmariadb (missing: unofficial-libmariadb_DIR)
-- MySQL Include dir: /usr/include/mysql  library dir: /usr/lib/x86_64-linux-gnu
-- MySQL client libraries: /usr/lib/x86_64-linux-gnu/libmysqlclient.so;-ldl
-- IPO / LTO enabled
-- Configuring done
CMake Error in CMakeLists.txt:
  Target "tfs" requires the language dialect "CXX20" (with compiler
  extensions).  But the current compiler "GNU" does not support this, or
  CMake does not know the flags to enable it.


CMake Error in src/CMakeLists.txt:
  Target "tfslib" requires the language dialect "CXX20" (with compiler
  extensions).  But the current compiler "GNU" does not support this, or
  CMake does not know the flags to enable it.


CMake Error at src/CMakeLists.txt:173 (target_link_libraries):
  Target "tfslib" links to:

    pugixml::pugixml

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.`
```
On **Ubuntu 20.04** I got the same compilation error as Fernando:

```
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.6 LTS
Release:	20.04
Codename:	focal
```

```
../forgottenserver/build# cmake ..
-- Could NOT find cryptopp (missing: cryptopp_DIR)
-- Could NOT find unofficial-libmariadb (missing: unofficial-libmariadb_DIR)
-- MySQL Include dir: /usr/include/mysql  library dir: /usr/lib/x86_64-linux-gnu
-- MySQL client libraries: /usr/lib/x86_64-linux-gnu/libmysqlclient.so;-ldl
-- IPO / LTO enabled
-- Configuring done
CMake Error at CMakeLists.txt:76 (add_executable):
  Target "tfs" links to target "pugixml::pugixml" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at src/CMakeLists.txt:172 (add_library):
  Target "tfslib" links to target "pugixml::pugixml" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by ranisalt** [comment]
This is because the pugixml CMake module shipped with Ubuntu 20 is too old and was significantly changed. If you can, please use Ubuntu 22 instead.

If you can't upgrade, I recommend using `vcpkg` with the following steps:

1. Clone the vcpkg repo: `git clone https://github.com/Microsoft/vcpkg.git`. Make sure you are in the directory you want the tool installed to before doing this (if in doubt, I recommend installing in the home folder).
2. Run the bootstrap script to build vcpkg: `./vcpkg/bootstrap-vcpkg.sh`
3. Generate CMake files using the vcpkg preset: `cmake --preset ninja-vcpkg`
4. Build using `cmake --build --preset ninja-vcpkg`, or by first `cd`ing to the `build` directory and then running `ninja` (not `make`)

If `ninja` is not installed, please install it using `apt install ninja-build`. You do not need to install the rest of the dependencies using `apt`, just `git cmake build-essential ninja-build`.

**Comment by fernandorrmartins** [comment]
> This is because the pugixml CMake module shipped with Ubuntu 20 is too old and was significantly changed. If you can, please use Ubuntu 22 instead.
> 
> If you can't upgrade, I recommend using `vcpkg` with the following steps:
> 
> 1. Clone the vcpkg repo: `git clone https://github.com/Microsoft/vcpkg.git`. Make sure you are in the directory you want the tool installed to before doing this (if in doubt, I recommend installing in the home folder).
> 2. Run the bootstrap script to build vcpkg: `./vcpkg/bootstrap-vcpkg.sh`
> 3. Generate CMake files using the vcpkg preset: `cmake --preset ninja-vcpkg`
> 4. Build using `cmake --build --preset ninja-vcpkg`, or by first `cd`ing to the `build` directory and then running `ninja` (not `make`)
> 
> If `ninja` is not installed, please install it using `apt install ninja-build`. You do not need to install the rest of the dependencies using `apt`, just `git cmake build-essential ninja-build`.

I instaled virtual box in my machine, installed ubuntu 22, and tried to compile again, and i got this problem.
![image](https://user-images.githubusercontent.com/32397071/235371705-36dbf07b-fda6-4888-87fd-b7d6b5bdd043.png)

**Comment by ranisalt** [comment]
Just created a new Ubuntu 22.04 machine and followed instructions, the build works fine. You seem to be missing `libpugixml-dev` which provides `/usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake`.

**Comment by fernandorrmartins** [comment]
> Just created a new Ubuntu 22.04 machine and followed instructions, the build works fine. You seem to be missing `libpugixml-dev` which provides `/usr/lib/x86_64-linux-gnu/cmake/pugixml/pugixml-config.cmake`.
> 
> Also note that you are running an extremely outdated version of gcc (7.5.0) which may not be enough to build the source code. An `apt update` + `apt upgrade` should help.

I will try to install this libpugixml-dev, but i already did an apt-update + apt upgrade, and the gcc is that one.
I tried to compile the code in Ubuntu 18, and Ubuntu 22. I getting error in this two versions.
I tried to use WSL2 in windows to run this two versions of Ubuntu, Virtual Box, and my cloud of Ubuntu, and got the same error in both.

I think i already instaled PugiXML but seems the problem is with environment paths. I dont know if i need to configure it, and i dont know how to do it.

About this steps you gave to me, i dont know where to run any comand, if is in the folder of the server. I think maybe need more steps.
![image](https://user-images.githubusercontent.com/32397071/235489583-a034a649-2926-4819-a9f0-16388b42a6d3.png)

**Comment by Mau005** [comment]
I have tried anyway how to get it resolved and still have problems, I managed to compile
but I have alerts and I have a server error.

```
root@ainhosoft:/home/forgottenserver/build# cmake -DCMAKE_TOOLCHAIN_FILE=/home/vcpkg/scripts/buildsystems/vcpkg
.cmake ..
-- Running vcpkg install
-- Running vcpkg install - done
-- The CXX compiler identification is GNU 9.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Crypto++: /home/forgottenserver/build/vcpkg_installed/x64-linux/include
-- Found ZLIB: /home/forgottenserver/build/vcpkg_installed/x64-linux/lib/libz.a (found version "1.2.13")
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found OpenSSL: /home/forgottenserver/build/vcpkg_installed/x64-linux/debug/lib/libcrypto.a
-- Found LuaJIT: /usr/lib/x86_64-linux-gnu/libluajit-5.1.so;/usr/lib/x86_64-linux-gnu/libm.so (found version "2.1.0-beta3")
CMake Warning at /usr/share/cmake-3.16/Modules/FindBoost.cmake:1161 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1283 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1921 (_Boost_MISSING_DEPENDENCIES)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  /home/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:64 (find_package)


CMake Warning at /usr/share/cmake-3.16/Modules/FindBoost.cmake:1161 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1283 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1921 (_Boost_MISSING_DEPENDENCIES)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  /home/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:64 (find_package)


CMake Warning at /usr/share/cmake-3.16/Modules/FindBoost.cmake:1161 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1283 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:1921 (_Boost_MISSING_DEPENDENCIES)
  build/vcpkg_installed/x64-linux/share/boost/vcpkg-cmake-wrapper.cmake:11 (_find_package)
  /home/vcpkg/scripts/buildsystems/vcpkg.cmake:806 (include)
  CMakeLists.txt:64 (find_package)


-- Found Boost: /home/forgottenserver/build/vcpkg_installed/x64-linux/include (found suitable version "1.82.0", minimum required is "1.66.0") found components: system iostreams regex
-- IPO / LTO enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /home/forgottenserver/build
```

the error in tfs is:
the server works but it crashes, it can be in 5 minutes or in 10 hours.

```
terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
  what():  Invalid argument [system:22]
Aborted
```

installed dependencies
`./vcpkg install boost-iostreams:x64-linux boost-asio:x64-linux boost-system:x64-linux boost-filesystem:x64-linux boost-variant:x64-linux boost-lockfree:x64-linux luajit:x64-linux libmariadb:x64-linux pugixml:x64-linux mpir:x64-linux cryptopp:x64-linux`


Even though this library I couldn't manage to install it:
`mpir:x64-linux`

**Comment by EPuncker** [comment]
what is the status of this issue as of latest tfs?

**Comment by telania** [comment]
[100%] Linking CXX executable tfs
lto-wrapper: warning: using serial compilation of 6 LTRANS jobs


My error

**Comment by Nextness** [comment]
So, this took me a while to get to work. I had the same issue as @fernandorrmartins. For some context, I'm using WSL2 - Ubuntu 22.04.3 LTS. First I followed the steps in the documentation and installed the required programs. Following that I did what was in the explanation described here https://otland.net/threads/tfs-1-5-compilation.287315/:
```bash
sudo apt install --reinstall binutils
sudo apt install --reinstall liblua5.4-0
sudo apt install --reinstall liblua5.4-dev
sudo apt install --reinstall g++-10
sudo apt update
sudo apt upgrade
```
**I don't actually know if these steps are actually required, but just for the sake of documenting every step.**

Then I installed `libboost-locale-dev` with the command `apt install libboost-locale-dev`. After that I was able to `cmake ..` and then compile. Hopefully this helps.

Error
![image](https://github.com/otland/forgottenserver/assets/10326570/44a1138b-0639-4892-b074-3509047e745b)

After installing `libboost-locale-dev`
![image](https://github.com/otland/forgottenserver/assets/10326570/5164e9ac-a97e-4fda-b2cd-8b64e9baaffb)

After `make`
![image](https://github.com/otland/forgottenserver/assets/10326570/c48ddb68-5461-4f5a-83eb-e1278eb25b58)

P.S.: Also also, I did some testing; in Ubuntu 20.04 LTS I had to manually update cmake and openssl to the correct versions, to then install `libboost-locale-dev`.
